decidim-admin/app/controllers/concerns/decidim/admin/needs_admin_tos_accepted.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Admin
+    # Shared behaviour for signed_in admins that require the latest TOS accepted
+    module NeedsAdminTosAccepted
+      extend ActiveSupport::Concern
+
+      included do
+        before_action :tos_accepted_by_admin
+      end
+
+      private
+
+      def tos_accepted_by_admin
+        return unless request.format.html?
+        return unless current_user
+        return if current_user.admin_terms_accepted?
```

Comment:
I'm probably missing something about what you're proposing here, but as far as I understand adding this guard clause in this case isn't necessary, and in fact would re-introduce this bug for roles that have access to the admin panel (and need to accept the admins' TOS) even though they don't have the admin boolean attribute configured. 
This is the case of assembly's admins, valuators, collaborators, moderators, etc. 

I've added one of these roles in the specs to make it more clear (see 187f48ee3c8334af7a67b9f26d273479f983dc27). 

Let me know if there's something that I'm missing though!

Reply:
That's true, didn't think about that.

The bug that I was facing was:
1. Login as `user@example.org`
2. Type in the URL `/admin/templates/questionnaire_templates`
3. Get redirected to the admin TOS page which gives you a 404

But I can also see this as acceptable since I would not expect most users try something like that (it's just me :smile:).